### DISCUSSION TITLE
Setting up ESP32 with multiple clients on different protocols

### DISCUSSION THREAD

**Discussion by geeks-r-us** [eModbus/eModbus#286] [Q&A]
Hi, I have following setup

RTU server <--> ESP32 <--> TCP client

and I want to use the RTU server from inside of the ESP32 program as well as provide access to the device via TCP client.
Usage: while normal usage ESP32 queries data and publishes it to mqtt. But when I want to configure the device via vendor software it needs to connect via TCP client. 

Can I use ModbusBridgeWifi for this scenario? Like in the example: [Wifi RTU Bridge](https://github.com/eModbus/eModbus/blob/master/examples/WiFi-RTU_Bridge/main.cpp)?

How can I extend this example with local queries from the ESP32 code?

Do I need to stop the local queries from ESP32 to make TCP client work?
If yes how can I detect that a TCP client has connected?

Thank you.

**Comment by Miq1** [comment]
That scenario is exactly what a bridge is intended to do. To the TCP side it acts as a regular TCP server, while on the RTU side it is a client. 
Regardless if the requests are coming in via TCP or are issued by a ``localRequest()`` on the ESP32 itself call they will be handled identically. So normally the ESP will do these local requests in ``loop()`` and process the responses right there (local requests will be synchronous), while the TCP requests are handled in the background automatically. 
To request the ESP32 itself you may add a locally chosen server ID and processing functions for any function codes to the bridge. If now some TCP request will be addressed to this server ID, your functions will be called like with any plain TCP server.
